Add initial feature set

mattburgess opened the pull request.

Commit:
Add initial feature set

Commit:
Add CI job and GitHub repo config

dan-hill2802 commented on vpc_endpoints.tf:
```diff
@@ -0,0 +1,96 @@
+  tags        = "${var.common_tags}"
+}
+
+resource "aws_security_group_rule" "vpce_kafka_https_ingress" {
```
kafka shouldn't be in the resource name

dan-hill2802 commented on vpc_endpoints.tf:
```diff
@@ -0,0 +1,96 @@
+  security_group_id        = "${aws_security_group.vpc_endpoints.id}"
+}
+
+resource "aws_security_group_rule" "kafka_vpce_https_egress" {
```
kafka shouldn't be in the resource name

dan-hill2802 commented on variables.tf:
```diff
@@ -0,0 +1,90 @@
+}
+
+variable "region" {
+  description = "(Optional) The region in which to deploy the VPC."
```
marked optional, but no default. Maybe grab the caller id's region?

mattburgess replied:
Fixed by removing the optional label.

Commit:
Simplify variable descriptions

The required/optional status of a variable and its default status are
documented automatically by Terraform Registry

Commit:
Make SG rule names generic
